django/thunderstore/api/cyberstorm/views/packages.py

```diff
@@ -0,0 +1,467 @@
+
+        return (
+            self._get_base_queryset()
+            .active()  # type: ignore
+            .prefetch_related(
+                "community_listings__categories",
+                "community_listings__community",
+            )
+            .select_related("latest", "namespace")
+            .annotate(
+                download_count=Subquery(
+                    this_package.annotate(
+                        downloads=Sum("versions__downloads"),
+                    ).values("downloads"),
+                ),
+            )
+            .annotate(
+                rating_count=Subquery(
+                    this_package.annotate(
+                        ratings=Count("package_ratings"),
+                    ).values("ratings"),
+                ),
+            )
+        )
+
+    def _filter_deprecated(
+        self,
+        show_deprecated: bool,
+        queryset: QuerySet[Package],
+    ) -> QuerySet[Package]:
+        """
+        Deprecated packages are included only if specifically requested.
+        """
+        if show_deprecated:
+            return queryset
+
+        return queryset.exclude(is_deprecated=True)
+
+    def _filter_nsfw(
+        self,
+        show_nsfw: bool,
+        queryset: QuerySet[Package],
+    ) -> QuerySet[Package]:
+        """
+        NSFW packages are included only if specifically requested.
+        """
+        if show_nsfw:
+            return queryset
+
+        return queryset.exclude(community_listings__has_nsfw_content=True)
+
+    def _filter_by_categories(
+        self,
+        params: OrderedDict,
+        queryset: QuerySet[Package],
+    ) -> QuerySet[Package]:
+        """
+        Include only packages (not) belonging to specific categories.
+
+        Multiple categories are OR-joined, i.e. if included_categories
+        contain A and B, packages belonging to either will be returned.
+        """
+        if params["included_categories"]:
+            queryset = queryset.exclude(
+                ~Q(
+                    community_listings__categories__slug__in=params[
+                        "included_categories"
+                    ],
+                ),
+            )
+
+        if params["excluded_categories"]:
+            queryset = queryset.exclude(
+                community_listings__categories__slug__in=params["excluded_categories"],
+            )
+
+        return queryset
+
+    def _filter_by_section(
+        self,
+        section_slug: Optional[str],
+        queryset: QuerySet[Package],
+    ) -> QuerySet[Package]:
+        """
+        PackageListingSections can be used as shortcut for multiple
+        category filters.
+        """
+        if not section_slug:
+            return queryset
+
+        try:
+            section = PackageListingSection.objects.prefetch_related(
+                "require_categories",
+                "exclude_categories",
+            ).get(slug=section_slug)
```

Comment:
The slug field is only unique in combination with the community, so you need to query based on both. This would crash in production as it is currently written

Reply:
Honestly I'd just use the UUID field instead (which is immutable & unique). The only downside is loss of human readability, but that shouldn't be a concern in this case at all.